funky-flamingos/rinky_dink/users/views.py

```diff
@@ -0,0 +1,199 @@
+            messages.success(
+                request, f"You have been successfully added to {team.name}!"
+            )
+            return redirect("dashboard")
+        else:
+            messages.error(
+                request,
+                "Unable to add you to team! Please recheck team name and password",
+            )
+
+    return render(request, "users/team_join.html")
+
+
+def login(request):
+    if request.method == "POST":
+        username = request.POST["username"]
+        password = request.POST["password"]
+        user = auth.authenticate(username=username, password=password)
+        if user:
+            auth.login(request, user)
+            messages.success(request, "You have successfully logged in")
+            return redirect("dashboard")
+        else:
+            messages.error(request, "Invalid Credentials")
+            return redirect("login")
+    return render(request, "users/login.html")
+
+
+@login_required(login_url="/login/")
+def upload(request):
+    context = {}
+    if request.method == "POST":
+        uploaded_file = request.FILES["document"]
+        if not check_title_presence(uploaded_file):
+            messages.success(request, "New file save successfully.")
+            path = save_on_server(uploaded_file)
+            hash_val = hash_file(path)
+            form = File(title=uploaded_file.name, hash_val=hash_val, user=request.user)
+            form.save()
+            file_group_form = FileGroup(title=uploaded_file.name, user=request.user)
+            file_group_form.save()
+            file_group_form.files.add(form)
+        else:
+            # flash message that A similar file is already present  if hash value matches
+            path = save_on_server(uploaded_file)
+            hash_val = hash_file(path)
+            if check_file_presence(hash_val):
+                messages.warning(request, "A similar file is already present")
+                os.remove(path)
+
+            # hash value matches so no need to upload. if not, upload it with different file name
+            else:
+                file_group = FileGroup.objects.filter(title=uploaded_file.name)[0]
+                # save file on server with new name{version}
+                title = path.split("/")[-1]
+                form = File(title=title, hash_val=hash_val, user=request.user)
+                form.save()
+                file_group.files.add(
+                    form
+                )  # adds newly created file version to file group
+                messages.warning(
+                    request,
+                    "file with same name is already present adding it with new version ",
+                )
+
+    return render(request, "users/upload.html", context)
+
+
+@login_required(login_url="/login/")
+def dashboard(request):
+    teams = Member.objects.filter(user__id__in=[request.user.id])
+    context = {}
+    files = FileGroup.objects.filter(user__id__in=[request.user.id])
```

Comment:
Similar things look better together. Here, if the first and the third line were next to each other and the `context = {}` was the third line, this would look much tidier.

More importantly, you're also redefining the `context` variable on the next line, so creating it as an empty dictionary does exactly nothing, and line 139 here should be completely removed.